src/sage/rings/finite_rings/residue_field.pyx

```diff
@@ -1436,37 +1529,41 @@ cdef class LiftingMap(Section):
-            sage: F.lift_map()                                                          # optional - sage.rings.number_field
+            sage: # needs sage.rings.number_field
+            sage: K.<theta_5> = CyclotomicField(5)
+            sage: F = K.factor(7)[0][0].residue_field()
+            sage: F.lift_map()
             Lifting map:
               From: Residue field in theta_5bar of Fractional ideal (7)
               To:   Maximal Order in Cyclotomic Field of order 5 and degree 4
 
+            sage: # needs sage.rings.number_field
             sage: x = polygen(ZZ, 'x')
-            sage: K.<a> = NumberField(x^5 + 2)                                          # optional - sage.rings.number_field
-            sage: F = K.factor(7)[0][0].residue_field()                                 # optional - sage.rings.number_field
-            sage: L = F.lift_map(); L                                                   # optional - sage.rings.number_field
+            sage: K.<a> = NumberField(x^5 + 2)
+            sage: F = K.factor(7)[0][0].residue_field()
+            sage: L = F.lift_map(); L
             Lifting map:
               From: Residue field in abar of Fractional ideal (2*a^4 - a^3 + 4*a^2 - 2*a + 1)
               To:   Maximal Order in Number Field in a
                     with defining polynomial x^5 + 2
-            sage: L.domain()                                                            # optional - sage.rings.number_field
+            sage: L.domain()
             Residue field in abar of Fractional ideal (2*a^4 - a^3 + 4*a^2 - 2*a + 1)
 
-            sage: K.<a> = CyclotomicField(7)                                            # optional - sage.rings.number_field
-            sage: F = K.factor(5)[0][0].residue_field()                                 # optional - sage.rings.number_field
-            sage: L = F.lift_map(); L                                                   # optional - sage.rings.number_field
+            sage: # needs sage.rings.number_field
+            sage: K.<a> = CyclotomicField(7)
+            sage: F = K.factor(5)[0][0].residue_field()
+            sage: L = F.lift_map(); L
             Lifting map:
               From: Residue field in abar of Fractional ideal (5)
               To:   Maximal Order in Cyclotomic Field of order 7 and degree 6
-            sage: L.codomain()                                                          # optional - sage.rings.number_field
+            sage: L.codomain()
             Maximal Order in Cyclotomic Field of order 7 and degree 6
 
+            sage: # needs sage.rings.finite_rings
             sage: R.<t> = GF(2)[]; h = t^5 + t^2 + 1
             sage: k.<a> = R.residue_field(h)
             sage: K = R.fraction_field()
-            sage: L = k.lift_map(); L.codomain()
+            sage: L = k.lift_map(); L.codomain()                                        # needs sage.libs.ntl
             Univariate Polynomial Ring in t over Finite Field of size 2 (using GF2X)
```

Comment:
If `sage.libs.ntl` is needed just for `k.lift_map()` and `L.codomain()`, perhaps `sage.rings.finite_rings` should depend on `sage.libs.ntl`. No?

Reply:
No, the test also works when only PARI provides an implementation, but then the text "(using GF2X)" is missing.

Reply:
I see. 

An alternative (but less confusing) way to handle it is
```
            sage: L = k.lift_map(); L.codomain()                                       
            Univariate Polynomial Ring in t over Finite Field of size 2 (...)
```
or something similar.

Reply:
OK.

Reply:
Thanks.